Added sympify translators for basic numpy datatypes(Fixes #11086)

ArifAhmed1995 opened the pull request:
This is reference to [#11086](https://github.com/sympy/sympy/issues/11086)

Previously for 64-bit systems running Python 3.x numpy's int64 was converted into float.This fix converts it to 'int'.

UPDATE : sympify translators for basic numpy datatypes have been added.Unit tests have also been written.

Previously :

```
import numpy as np
import sympy as sp

x = sp.symbols('x')
print(x * 1)
print(x * np.int64(1))

```

Output : 

```
x
1.0*x

```

```
With the fix : 
import numpy as np
import sympy as sp

x = sp.symbols('x')
print(x * 1)
print(x * np.int64(1))
```

Output : 

```
x
x
```

ArifAhmed1995 committed:
sympify.core : numpy's int64 correctly converted to int

On 64-bit systems running Python 3.x , numpy's 'int64' was incorrectly
converted into 'float'.This simple fix resolves the issue and now
'int64' convertes into 'int' for both Python2/3 on 32/64 bit systems.

moorepants commented:
Please add a unit test for this change. And you have a test failure:

```
 File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/sympy-1.0.1.dev0-py2.7.egg/sympy/core/tests/test_sympify.py", line 371, in test_int_float

    assert isinstance(sympify(i5b), Float)

AssertionError
```

ArifAhmed1995 commented:
@moorepants Yes , I'll definitely run tests locally before pushing from now on.If it passes I'll add the unit test.

ArifAhmed1995 committed:
sympify : Add support for translating basic numpy datatypes

Basic numpy datatypes are translated correctly by sympify.
They are :
1.bool_
2.int_,intc,intp,int8,int16,int32,int64
3.uint8,uint16,uint32,uint64
4.float_,float16,float32,float64
5.complex_,complex64,complex128

ArifAhmed1995 committed:
sympify : Add support for translating basic numpy datatypes

Basic numpy datatypes are translated correctly by sympify.
They are :
1.bool_
2.int_,intc,intp,int8,int16,int32,int64
3.uint8,uint16,uint32,uint64
4.float_,float16,float32,float64
5.complex_,complex64,complex128

Deleted an unecessary comment.

ArifAhmed1995 committed:
sympify : Add support for translating basic numpy datatypes

Add a if-condition to make sure that only <numpy.number>
is attempted to translate.

ArifAhmed1995 committed:
Merge branch 'master' of https://github.com/sympy/sympy into core.sympify

ArifAhmed1995 committed:
sympify:Add support for translating basic numpy datatypes

Add an if-condition to make sure only <numpy.number>
is attempted to translate

ArifAhmed1995 committed:
Merge branch 'core.sympify' of https://github.com/ArifAhmed1995/sympy into core.sympify

Conflicts:
	sympy/core/sympify.py

sympify : Add support for translating basic numpy datatypes

Add an if-condition to make sure only <numpy.number> is
attempted to be translated.

ArifAhmed1995 commented:
@moorepants @asmeurer 

There are two tests failing now : 

1.Code quality : I've rectified this by running `./bin/strip_whitespace -r sympy`.

2.The following error as reported by TravisCI : 
```
___________ sympy/core/tests/test_relational.py:test_new_relational ____________
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/sympy-1.0.1.dev0-py3.4.egg/sympy/core/tests/test_relational.py", line 272, in test_new_relational
    raises(ValueError, lambda: Relational(x, 1, relation_type))
  File "/home/travis/virtualenv/python3.4.2/lib/python3.4/site-packages/sympy-1.0.1.dev0-py3.4.egg/sympy/utilities/pytest.py", line 81, in raises
    raise AssertionError("DID NOT RAISE")
AssertionError: DID NOT RAISE
```
I have run `./bin/test test_relational` and `./bin/test pytest` in virtual environment Python - 3.4 as explained on the Development Workflow page.On local machine no errors are reported but TravisCI reports failures.

Can someone help me here ?
